app/src/main/java/rick_and_morty/ui/characters/CharactersViewModel.kt

```diff
@@ -17,33 +15,37 @@ class CharactersViewModel @Inject constructor(
     private val characterRepository: CharacterRepository
 ) : ViewModel(){
 
-    private var initialLoad: Boolean = true
     private var page = 1;
 
-    private var _characters = MutableStateFlow<List<CharacterResultsDto>>(emptyList())
-    val characters: StateFlow<List<CharacterResultsDto>> = _characters
+    private var _characters = MutableStateFlow(UIState())
+    val characters: StateFlow<UIState> = _characters
+
+    private var characterList = mutableListOf<CharacterResultsDto>()
```

Comment:
let's remove this. `_characters` already contains the list of displayed characters

Reply:
Done